src/database/migrations/0012-create-platform_provider_responses.sql

```diff
@@ -0,0 +1,12 @@
+CREATE TABLE platform_provider_responses (
+  id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  external_id VARCHAR NOT NULL,
+  platform_provider VARCHAR NOT NULL,
```

Comment:
Are we wanting to keep a history? I don't think it'll be useful, and in the spirit of light-weightedness, may I instead propose:

```suggestion
  external_id VARCHAR,
  platform_provider VARCHAR,
  PRIMARY KEY (external_id, platform_provider),
```

In particular, that should make retrieval easier, as the client can assume one-or-none rather than having to pick through an array.

Reply:
I'm not sure we've spent enough time with this problem space to be able to confidently rule out history use cases as being a desired feature.  Most signs have pointed to "we always want an audit trail" so I would start off assuming we *DO* want history in the system.

On another note: I was under the impression that not having an surrogate primary key is an anti pattern, but that might be an out of date concept (I believe it's rooted in DB optimizations / faster joins).  This is to say, we could create a UNIQUE constraint on external_id and platform_provider as a natural key but I'd be wary to lose the `id` field.

Were you planning on doing one API call per platform provider (as opposed to just getting all the data in one call)?

Reply:
... actually, maybe the solution here is to not use external ID in the first place, and instead store this in relation to an applicant via an `applicant_id` foreign key.

Reply:
> On another note: I was under the impression that not having an surrogate primary key is an anti pattern, but that might be an out of date concept (I believe it's rooted in DB optimizations / faster joins). This is to say, we could create a UNIQUE constraint on external_id and platform_provider as a natural key but I'd be wary to lose the `id` field.

My understanding is the opposite: surrogate keys are only useful if the natural keys are very large, or likely to change, and should otherwise be avoided. [Josh Berkus's post Primary Keyvil](http://www.databasesoup.com/2015/03/primary-keyvil-reprised.html) is a great read; I strongly recommend it!

Reply:
@slifty and I just talked! I'll try to summarize our conversation.

My main concern was that as a consumer of the read API, I didn't want to have to pick among multiple versions of data platform provider responses. So long as the query only returns one-or-zero for each, that concern is addressed, and I don't have a strong opinion on keeping multiple versions.

The benefit of keeping multiple versions would be to have the option to someday show a history; that's something we baked in very deeply to proposals, and it's not unreasonable to think we might want this here.

However, the framing I think we arrived at is that this is a _cache_ of Charity Navigator / Candid's data, and nothing more. If we decide we want a history of that data in the PDC, then that will be one (of several) reasons to throw out this implementation. We had considered - and rejected - the option of being a proxy to those services, primarily because we didn't want our availability to be dependent on their availability, and secondarily because we didn't want that code to live in this repo; a cache - with the code living in the data-scripts repo - is a more conservative approach that resolves both of those concerns.